src/extent.c

```diff
@@ -1224,7 +1224,8 @@ extent_record(tsdn_t *tsdn, arena_t *arena, ehooks_t *ehooks, ecache_t *ecache,
 			    &arena->edata_cache, ehooks, rtree_ctx, ecache,
 			    edata, &coalesced, growing_retained);
 		} while (coalesced);
-		if (edata_size_get(edata) >= oversize_threshold) {
+		if (edata_size_get(edata) >= oversize_threshold &&
+		    arena_may_force_decay(arena)) {
```

Comment:
Not quite happy with this getting more arbitrary. However it's still useful to avoid confusion (people reporting purging happens on regular arenas, instead of huge arena, even with decay off).